python3eval/pythoneval.cs

```diff
@@ -63,6 +64,14 @@ public override object Evaluate(string code, IList bindingNames, IList bindingVa
             if (code != prev_code)
             {
                 var pythonEngine = Python.CreateEngine();
+                //to maintain compatability with ironPython code written in .netframework - we load the system.dll shim
+                //which loads many types which used to be in system.dll(mscorlib) like system.diagnostics.process
+                //which were moved in .Net. These types are now importable by python code without users needing to add
+                //clr.addreference.
+                //TODO consider a preference for this.
+                //TODO test smaller shims...netstd.dll
+                pythonEngine.Runtime.LoadAssembly(Assembly.Load("System"));
+
```

Comment:
how did this work in netframework ?

Reply:
See the following snippet from the IronPython docs:
`
By default, core assemblies (mscorlib.dll and System.dll on .NET Framework / System.Private.CoreLib.dll on .NET Core) are automatically loaded by the DLR. This enables you to start using these assemblies (which IronPython itself is dependent on) without having to call clr.AddReference-methods.

`

I looked for where the DLR did this and even asked on the IronPython GitHub but didn't get a concrete line number.

Reply:
I mean…those assemblies still exist as part of dotnet8. They just do typeforwardind. Maybe the dlr (or iron pythons usage of it) does not handle typeforwarding very well?

Reply:
The DLR is a package, it is actually included in the IronPython repo as a git submodule (shudder)
https://github.com/IronLanguages/dlr/tree/872730ab305d37f67179db8ca778d1cd22984e67

if the docs are to be believed, the DLR on .Net loads and imports system.private.core which does not include the shims.

On .net48 the DLR loads system.dll where shims were not needed as all these types were in system.dll and so they were imported.

By importing System.dll now, we're loading the shim assembly, so loading the types which are forwarded, and then importing them into the python engine.

I think the python engine handles shims just fine, it's just the standard entry point which has changed.